app/api/upload/[type]/route.ts

```diff
@@ -0,0 +1,44 @@
+import { NextResponse } from "next/server";
+import { auth } from "@/lib/auth";
+import { createPresigned } from "@/lib/s3";
+
+export async function POST(
+  request: Request,
+  { params: { type } }: { params: { type: string } },
+): Promise<NextResponse> {
+  const params = await request.json();
+  const session = await auth();
+  if (!session?.user?.id)
+    return NextResponse.json(
+      { error: "Unauthenticated access" },
+      { status: 400 },
+    );
+
+  let uniqueKey = "";
+
+  if (type === "avatar") {
+    const { filename } = params;
+    const userId = Number(session.user.id);
+    uniqueKey = `user/${userId}/${filename}`;
+  } else if (type === "project-files") {
+    const { filename, projectId, specId } = params;
+    uniqueKey = `project/${projectId}/task/${specId}/${filename}`;
```

Comment:
these values should be validated to makesure they aren't undefined
but also the project and task spec should exist too

as is could allow for someone to upload a file to a project that doesn't exist yet, and when that project with that id is created that file would already exist for that task, could be like a fake legal document or a different design etc. not ideal